lib/libc/minimal/source/stdout/stdout_console.c

```diff
@@ -65,6 +65,10 @@ size_t z_impl_zephyr_fwrite(const void *_MLIBC_RESTRICT ptr, size_t size,
 		return 0;
 	}
 
+	if ((strlen(ptr) + 1) < (size * nitems)) {
```

Comment:
The `size` and `nitems` should be checked prior to be passed to `fwrite`, it is hard to predict where they are coming from and what is true meaning to values to these parameters inside the function.

`fwrite` should not limit buffer to closest 0, the puts does so; This will probably introduce bug that will be later discovered when someone will be, for example, catching console to file while writing binary to stdout.
In unix-like systems you are able to cat binary file to console and catch it through pipe by other process, for example:
```
cat bla.tgz | tar xvzf -
```
would not work if output would be cut at first 0.
Additionally the `strlen` may not solve problem when the `ptr` points to a long memory region with no 0 (empty flash for example), or there is some strange memory layout where address of region wraps around.

Reply:
Hi ,@de-nordic, thank you for reviewing it. And the example you provide here is very clear. I believe this is a wrong solution for fixing it. 

Could you also please let me know:  Do you think whether #33491 is not a bug? I have the same point as what @pabigot said and it should be closed.